bundle.c

```diff
@@ -1,68 +1,98 @@
-
-#include <stdio.h>
 #include <ctype.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-char* tovname(char* str) {
-    int length = strlen(str);
-    char* buf = malloc(length + 2);
-    buf[0] = '_';
-    for (int i = 0; i < length; ++i) {
-        if (isalnum(str[i])) buf[i+1] = toupper(str[i]);
-        else buf[i+1] = '_';
-    }
-    buf[length + 2] = '\0';
-    return buf;
+char *tovname(char *str) {
+  int length = strlen(str);
+  char *buf = malloc(length + 2);
+  buf[0] = '_';
+  for (int i = 0; i < length; ++i) {
+    if (isalnum(str[i]))
+      buf[i + 1] = toupper(str[i]);
+    else
+      buf[i + 1] = '_';
+  }
+  buf[length + 1] = '\0';
+  return buf;
 }
 
 #define LSTR "_LENGTH"
 #define LSTRL 7
 
-char* tovnamel(char* str) {
-    int length = strlen(str);
-    char* buf = malloc(length + 2 + LSTRL);
-    buf[0] = '_';
-    for (int i = 0; i < length; ++i) {
-        if (isalnum(str[i])) buf[i+1] = toupper(str[i]);
-        else buf[i+1] = '_';
-    }
-    for (int i = 0; i < LSTRL; ++i) {
-        buf[length + 1 + i] = LSTR[i];
-    }
-    buf[length + 2 + LSTRL] = '\0';
-    return buf;
+char *tovnamel(char *str) {
+  int length = strlen(str);
+  char *buf = malloc(length + 2 + LSTRL);
+  buf[0] = '_';
+  for (int i = 0; i < length; ++i) {
+    if (isalnum(str[i]))
+      buf[i + 1] = toupper(str[i]);
+    else
+      buf[i + 1] = '_';
+  }
+  for (int i = 0; i < LSTRL; ++i) {
+    buf[length + 1 + i] = LSTR[i];
+  }
+  buf[length + 2 + LSTRL] = '\0';
+  return buf;
+}
+
+void print_usage() {
+  printf("Usage: ./bundle <file> [<file> <file>...] 2> file.c\n");
+  printf("  -h, --help      Show this help message and exit.\n");
 }
 
-int main(int argc, char** argv) {
-    int files = argc-1;
-    for (int i = 0; i < files; ++i) {
-        char* file = argv[i+1];
-        char* vname = tovname(argv[i+1]);
-        char* vnamel = tovnamel(argv[i+1]);
-        printf("Bundling %s as\n\t%s and %s\n", file, vname, vnamel);
-        FILE* f = fopen(file, "rb");
-        fseek(f, 0, SEEK_END);
-        long length = ftell(f);
-        fseek(f, 0, SEEK_SET);
-        unsigned char* buf = malloc(length);
-        fread(buf, 1, length, f);
+int main(int argc, char **argv) {
+  if (argc == 2 &&
+      (strcmp(argv[1], "-h") == 0 || strcmp(argv[1], "--help") == 0)) {
+    print_usage();
+    return 0;
+  }
+
+  int files = argc - 1;
+  if (files == 0) {
+    fprintf(stderr, "Error: No files specified.\n");
```

Comment:
I added this file guard in so that the user doesn't get confused on thinking to run `./bundle` instead of `./bundle [file1] ...`